test/state/mpt.hpp

```diff
@@ -13,6 +13,13 @@ constexpr auto emptyMPTHash =
 
 /// Insert-only Merkle Patricia Trie implementation for getting the root hash
 /// out of (key, value) pairs.
+///
+/// Limitations:
+/// 1. The keys must not be longer than 32 bytes.
```

Comment:
Is this possible to have these limitations as explicut asserions in `insert()` ?

Reply:
The first 3 are protected by asserts, but not at the beginning of the `insert()` (as you need to traverse the tree to find out). However, I can consistently add messages to these asserts and document these here.

There simply is not method for the 4th restriction.

Reply:
We can also add "death" tests but felt like overkill to me.